geometry/optimization/iris.cc

```diff
@@ -399,18 +399,19 @@ void MakeGuessFeasible(const HPolyhedron& P, const IrisOptions& options,
                        const VectorXd& closest, Eigen::VectorXd* guess) {
   const auto& A = P.A();
   const auto& b = P.b();
-  const int N = A.rows();
-  if (A.row(N - 1) * *guess - b[N - 1] > 0) {
-    // The HPolyhedron uniform sampler wants feasible points.  First
-    // try projecting the closest point back into the set.
-    *guess = closest - options.configuration_space_margin *
-                           A.row(N - 1).normalized().transpose();
-    // If this causes a different constraint to be violated, then just
-    // go back to the "center" of the set.
-    if (!P.PointInSet(*guess)) {
-      *guess = P.ChebyshevCenter();
+  const int M = A.rows();
+  // Try projecting the guess onto any violated constraints, one-by-one.
+  for (int i = M - 1; i >= 0; --i) {
+    if (A.row(i) * *guess - b[i] > 0) {
+      guess->noalias() = closest - options.configuration_space_margin *
+                                       A.row(i).normalized().transpose();
     }
   }
+  // If this causes a different constraint to be violated, then just return the
+  // Chebyshev center.
+  if (!P.PointInSet(*guess)) {
```

Comment:
Minor: There is an inconsistency here as there is no check if the Chebyshev ball radius is smaller than `configuration_space_margin`. In our example, I got few "near-singular" polytopes because the program continues after this line. 
I think we may want to fully give up and return `std::nullopt` in IRIS (becomes MaybeIRIS...) when you can not fit a reasonably sized ball inside it. Another option is to leave such a decision to user after IRIS.